SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestWorldComposition/SpatialTestWorldComposition.cpp

```diff
@@ -0,0 +1,178 @@
+				{
+					UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedTestActorBase::StaticClass(), FoundReplicatedBaseActors);
+
+					if (FoundReplicatedBaseActors.Num() == 1 && FoundReplicatedBaseActors[0]->IsA(AInitiallyDormantTestActor::StaticClass()))
+					{
+						if (FoundReplicatedBaseActors[0]->GetActorLocation().Equals(ActorsLocations[0], 10.0f))
+						{
+							FinishStep();
+						}
+					}
+				}
+			}, 10.0f);
+
+		// Move the Pawn to the TestLocation with index 1.
+		AddStepFromDefinition(ClientMovePawnStepDefinition, FWorkerDefinition::Client(1));
+
+		// Test that the ReplicatedActorLevel was loaded correctly and the IntiallyDormantActorLevel was unloaded.
+		AddStep(TEXT("SpatialTestWorldCompositionClientCheckTestLocation1"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+			{
+				// Check that the movement was correctly applied before checking the level loaded correctly.
+				if (ClientOnePawn->GetActorLocation().Equals(TestLocations[1], 1.0f))
+				{
+					UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedTestActorBase::StaticClass(), FoundReplicatedBaseActors);
+
+					if (FoundReplicatedBaseActors.Num() == 1 && FoundReplicatedBaseActors[0]->GetActorLocation().Equals(ActorsLocations[1], 10.0f))
+					{
+						FinishStep();
+					}
+				}
+			}, 10.0f);
+
+		// Move the Pawn to the TestLocation with index 2.
+		AddStepFromDefinition(ClientMovePawnStepDefinition, FWorkerDefinition::Client(1));
+
+		// Test that the ReplicatedAndNetLoadOnClientLevel was loaded correctly and the ReplicatedActorLevel was unloaded.
+		AddStep(TEXT("SpatialTestWorldCompositionClientCheckTestLocation2"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+			{
+				// Check that the movement was correctly applied before checking the level loaded correctly.
+				if (ClientOnePawn->GetActorLocation().Equals(TestLocations[2], 1.0f))
+				{
+					UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedTestActorBase::StaticClass(), FoundReplicatedBaseActors);
+
+					if (FoundReplicatedBaseActors.Num() == 1 && FoundReplicatedBaseActors[0]->GetActorLocation().Equals(ActorsLocations[2], 10.0f))
+					{
+						FinishStep();
+					}
+				}
+			}, 10.0f);
+
+		// Move the Pawn to the TestLocation with index 3.
+		AddStepFromDefinition(ClientMovePawnStepDefinition, FWorkerDefinition::Client(1));
+
+		// Test that the InitiallyDormantAndNetLoadLevel was loaded correctly and the ReplicatedAndNetLoadOnClientLevel was unloaded.
+		AddStep(TEXT("SpatialTestWorldCompositionClientCheckTestLocation3"), FWorkerDefinition::Client(1), nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+			{
+				// Check that the movement was correctly applied before checking the level loaded correctly.
+				if (ClientOnePawn->GetActorLocation().Equals(TestLocations[3], 1.0f))
```

Comment:
May be worth a small helper function for this, but not a huge deal

Reply:
By this, I mean the repeated if - getallactors -- check - check - finish

Reply:
Yeah that is correct, will do!